ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -140,15 +158,18 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, int maxConn
             connectionTracker.startTrackingInitiatedConnection(host, port, connId);
           }
         } else {
-          sends.add(new NetworkSend(connId, requestMetadata.requestInfo.getRequest(), null, time));
+          sends.add(new NetworkSend(connId, requestMetadata.requestInfo.getRequest(),
+              requestMetadata.clientNetworkRequestMetrics, time));
           connectionIdToRequestInFlight.put(connId, requestMetadata);
           iter.remove();
+          requestMetadata.onRequestDequeue();
         }
       } catch (IOException e) {
-        //@todo: add to a metric.
+        networkMetrics.networkClientIOError.inc();
         logger.error("Received exception while checking out a connection", e);
       }
     }
+    numPendingConnections.set(pendingRequests.size());
```

Comment:
so this is not necessary either - this metric should simply be updated at the end of `sendAndPoll`